src/validators/register.rs

```diff
@@ -1,6 +1,11 @@
 use regex::Regex;
 
-pub fn validate_username(username: String) -> (String, String) {
+use crate::repositories::user::{Field, UserRepository};
+
+pub async fn validate_username(
+    username: String,
+    user_repository: &mut UserRepository,
+) -> (String, String) {
```

Comment:
Tu by som asi radsej daval username referenciou a vracal Result, presne nieco taketo myslim bolo v slidoch ze by sme nemali robit

Reply:
Asi by stacil nejaky `Result<(), String>`

Reply:
Alebo to nechaj tak, ked pozeram na tu validation funkciu tak to by bol celkom otravny refactor :dd

Reply:
Ja to prave predtim zkousel pres result a vypadalo to fakt hnusne, ty totiz v obou pripadech musis vracet jak error_message tak tu value, aby se daly pouzit v tom templatu. Urcite by to slo udelat i s resultem ale realne ten error ti je k nicemu, stejne bys z nej jenom tahal message, tohle mi prislo hezci.

Reply:
jo mas pravdu, s tym Resultom by to bolo podstatne otravnejsie